lib/webdriveragent.js

```diff
@@ -80,12 +110,15 @@ class WebDriverAgent {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
+      this.setupNoSessionProxy();
       this.setupProxy(sessionId);
       return this.webDriverAgentUrl;
     }
 
     log.info('Launching WebDriverAgent on the device');
 
+    this.setupNoSessionProxy();
```

Comment:
Is it needed before line 165 (where we initialize the other proxy)? If not, I would vote for just putting the no-session proxy initialization into `setupProxy`.